Remove usages of jQuery

donaldwasserman opened the pull request:
note: I am waiting to update the tests [because of](https://github.com/EmberSherpa/ember-element-resize-detector/pull/17)

alexander-alvarez commented on addon/components/ember-scrollable.js:
```diff
@@ -228,8 +227,8 @@ export default Component.extend(InboundActionsMixin, DomMixin, {
   resizeScrollContent() {
-    const width = this.$().width();
-    const height = this.$().height();
+    const width = this.element.offsetWidth;
```
do you need the similar to https://github.com/EmberSherpa/ember-element-resize-detector/pull/17/files#diff-29d350929a33c5a7ef2514dcad5e9a52R14 ?

alexander-alvarez commented on addon/classes/scrollable.js:
```diff
@@ -12,11 +12,11 @@ export default class Scrollable {
 
   contentOuterSize() {
-    return this.contentElement[camelize(`outer-${this.sizeAttr}`)]();
+    return this.contentElement[`outer${capitalize(this.sizeAttr)}`];
```
Scrollbars are missing -- I think this needs to be changed to `offset`
![image](https://user-images.githubusercontent.com/11779993/48448218-4383de80-e76c-11e8-8d60-3fb8e12488f0.png)

http://youmightnotneedjquery.com/#outer_height

donaldwasserman commented:
@alexander-alvarez So I went through and tried best I could to compare a lot of the measurements between the jquery/non-jqery version. 

This appears to work for me locally, although I'm really not sure why the tests were passing, because it seems like they shouldn't.

alexander-alvarez commented on addon/components/ember-scrollable.js:
```diff
@@ -228,8 +227,8 @@ export default Component.extend(InboundActionsMixin, DomMixin, {
   resizeScrollContent() {
-    const width = this.$().width();
-    const height = this.$().height();
+    const width = this.element.offsetWidth;
```
I'm a little confused now, why did this one switch to `offset` and why did  https://github.com/alphasights/ember-scrollable/pull/98/files#diff-3fa203c13b3ee2249dcb66b55d4c297fR22 switch to `getHeight`?

Is that simply what you needed to do to get it working?

donaldwasserman replied:
I wish I could say I sat down and did the math to figure out all the details of why and how each piece interacted, but I basically just put a few console logs across the critical paths and both this branch and master, and this is the measurement that worked.

If you could try it out too, that would be great, but it does seem like it's working. 

I'd like to add another test to make sure the `onScrolledToBottom` action gets called, because that should have been failing, but wasn't. However, I'm not 100% sure how to get the elements to scroll.

alexander-alvarez replied:
I haven't forgotten about this, just haven't gotten around to it. Hopefully before or during this coming Thanksgiving holiday.

donaldwasserman commented:
Hey @alexander-alvarez I sort of lost track of this over the holidays and am now resuming my quest to remove jquery from Ember-light-table. 

What do you think we need to check to get those over the finish line?

alexander-alvarez commented:
I think I'd want some reasoning or extensive testing on many browsers to make sure this is a.o.k given the discrepancy mentioned above. This space is filled with subtitles that I am pretty unfamiliar with that I'm hesitant to make changes to it without knowing it will be g2g for our downstream consumers

donaldwasserman commented:
You know, I don’t remember a single thing about this PR.  Looking back through it, I’ll just redo everything with the getHeight replacement. 

I have an app that depends pretty heavily on ELT, so getting this done correctly is a high priority.

alexander-alvarez commented:
Apologies for delaying you @donaldwasserman

donaldwasserman commented:
@alexander-alvarez no prob! I was really motivated to get rid of jQuery from my app in december, but then lost interest when i realized how much i depended on liquid fire, lol. I'll hopefully have some time this week to get this wrapped up.

richard-viney commented:
@donaldwasserman Is the only thing remaining here the more. consistent use of the `getHeight` replacement. function?

donaldwasserman commented:
Hey @richard-viney - Thanks for the bump on this. I actually have this somewhat working locally. But the issue I'm having some outdated test dependencies. So I'm just trying to get that fixed up this morning and pushed up.

donaldwasserman commented:
@richard-viney, @alexander-alvarez  another quick update, I switched up some various things: 

- I added back jquery & non-jquery tests so we can have some certainty about the test coverage
- Now there's some failing tests that I can't quite figure out why, and have run out of time this AM to work on it. I'll pick it back up hopefully next week.

If anyone has any suggestions or notices a quick fix, let me know.

alvinvogelzang commented:
@donaldwasserman you accidentally changed `firstMovement` to `secondMovement` in `scroll-content-element-test.js`. I've fixed it in the following PR: https://github.com/donaldwasserman/ember-scrollable/pull/1

All the ember tests succeed locally but somehow the tests in Travis are failing but I'm not sure what to do about that.

donaldwasserman commented:
@alexander-alvarez - Looks like this should be good to go now, but i don't see travis reporting anything on here anymore. I know this is sort of off the radar for you, but any ideas? It would be nice to get this wrapped up, and desipite it taking me a year plus on this project, i'd be happy to take this over for you and get the outstanding PR's fixed up.

alexander-alvarez commented:
@donaldwasserman not sure what is going on with travis will have to take a closer look. I agree it would be nice to get this wrapped up & thank you for your continued contributions.

Also I'm in favor of moving this to an area where it gets more love. Unfortunately I'm not the project maintainer, and I don't know who is. It seems the listed maintainer left Alphasights a while ago @eugeniodepalo any ideas as to who succeeded you?

donaldwasserman commented:
@alexander-alvarez Not sure what the deal with Jenkins either, but it looks like there may be some random errors left:  https://travis-ci.com/donaldwasserman/ember-scrollable

rwwagner90 commented:
@donaldwasserman @alexander-alvarez anything I can do to help with these efforts? Ember-light-table, which uses ember-scrollable, is pretty much the only thing left in our app using jQuery.

donaldwasserman commented:
So the only error left is on the latest ember versions, and I think it’s an issue in the dummy app itself, but that’s just a wild guess. 

I’ll set aside some time on Monday to investigate

rwwagner90 commented:
@donaldwasserman where can I see the errors? Happy to take a look.

donaldwasserman commented:
If you want to check it out it’s right here: https://travis-ci.com/donaldwasserman/ember-scrollable

I don’t have access to this repo, so I don’t know why Travis has disappeared here

rwwagner90 commented:
@donaldwasserman ah, looks like it is saying the owner is undefined when it tries to use `.lookup` on it. Are the tests using the new testing syntax?

donaldwasserman commented:
Thanks @rwwagner90. I was trying to not touch any of the test code (since i'm just dropping in on this...uh...2 years ago)

I'll run the codemod on that see what happens

donaldwasserman commented:
Thanks for the pointer @rwwagner90 - things are working on my machine ™️ .

Here's travis: https://travis-ci.com/donaldwasserman/ember-scrollable/builds/144037283

rwwagner90 commented:
@donaldwasserman looks like everything is green! 🎉

donaldwasserman commented:
Thanks for the bump @rwwagner90. 

@alexander-alvarez  See #121 where i added github actions for tests

I'll squash all the commits in here after that gets merged so we get a clean test run, then I think we can talk about pushing this out & cutting a release for downstream

rwwagner90 commented:
Thanks for all your work on this @donaldwasserman!

donaldwasserman commented:
Hey @alexander-alvarez can you take a second to review + merge + maybe cut a new release?

eelke commented:
@donaldwasserman I tried out the latest [commit](https://github.com/donaldwasserman/ember-scrollable/commit/5972cb28b94876fa63f21a3b153fb45f08187410) from your feature branch in a project that I'm working on, but it doesn't work correctly: the drag handle cannot be dragged, and clicking on the scrollbar works neither. 

I just ran the dummy app from that same commit and confirmed that this problem exists there as well.

donaldwasserman commented:
thanks @eelke - thanks for pointing it out should be fixed now. Must have missed it some of the rebasing i did.

alexander-alvarez commented on addon/components/ember-scrollable.js:
```diff
@@ -198,11 +198,16 @@ export default Component.extend(InboundActionsMixin, DomMixin, {
+
+    let getters = {
+      'height': getHeight,
+      'weight': getWidth
```
```suggestion
      'width': getWidth
```

alexander-alvarez commented:
@donaldwasserman  thanks for all the work on this, the typo update in and side scrolling should work, hence this PR should finally be G2G

rwwagner90 commented:
Now that this is merged, can we release a new version and update ember-light-table?

alexander-alvarez commented:
We've released https://github.com/alphasights/ember-scrollable/releases/tag/v1.0.0
Thanks again @donaldwasserman  and everyone who's helped get this done

alexander-alvarez commented:
@donaldwasserman @rwwagner90  see https://github.com/offirgolan/ember-light-table/releases/tag/2.0.0-beta.5

donaldwasserman commented:
@alexander-alvarez Thanks a bunch! Glad we've finally got this across the finish line.
